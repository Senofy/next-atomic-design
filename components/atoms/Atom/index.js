import PropTypes from 'prop-types'
import styled from 'styled-components'
import { font, palette } from 'styled-theme'

export const Atom = styled.div`
    font-family: ${font('primary')};
    color: ${palette('grayscale', 0)};
`

Atom.propTypes = {
    reverse: PropTypes.bool,
}

Atom.defaultProps = {
    palette: 'grayscale',
}