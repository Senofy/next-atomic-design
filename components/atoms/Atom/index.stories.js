import { storiesOf } from '@storybook/react'
import { Atom } from './'

storiesOf('Atom', module)
    .add('default', () => (
        <Atom>Test</Atom>
    ))
