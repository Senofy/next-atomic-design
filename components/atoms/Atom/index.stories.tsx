import React from 'react'
import { Story, Meta } from '@storybook/react'

import { Atom } from '.'

export default {
	title: 'atoms/Atom',
	component: Atom,
	argTypes: {
		children: { control: 'text' },
	},
	args: {
		children: 'I am a div',
	},
} as Meta

const Template: Story = (args) => <Atom {...args} />

export const Normal = Template.bind({})
