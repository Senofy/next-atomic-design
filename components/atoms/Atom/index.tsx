import styled from 'styled-components';
import { color, fontFamily } from 'styled-system';


export const Atom = styled.div`
  ${fontFamily};
  ${color};
`;