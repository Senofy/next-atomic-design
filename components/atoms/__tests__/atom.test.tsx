import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Atom } from '../Atom'

afterEach(cleanup)

test('<Atom> should take a snapshot', () => {
	const { asFragment } = render(<Atom />)
	expect(asFragment()).toMatchSnapshot()
})

test('<Atom> renders children when passed in', () => {
	const { getByText } = render(<Atom>Hello</Atom>)
	expect(getByText(/hello/i)).toBeInTheDocument()
})
