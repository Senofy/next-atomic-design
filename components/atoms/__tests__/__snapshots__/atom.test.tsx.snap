// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Atom> should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="Atom-sc-1c1pzcy-0 hVnAoc"
  />
</DocumentFragment>
`;
